### PR TITLE
add notification of UPS creation

### PR DESCRIPTION
**Pull request by dwikler**
Hello Stuart,
I was able to subscribe the PPVS simulator you added with the TMS simulator.
I have modified the default port number of UPS Pull SCP to which the PPVS subscribes in order to match the default configuration of the TMS simulator.
I have also added in the TMS simulator the notification of creation of UPS to the AET which globally subscribed.

### CODE REVIEW

File: tdwii_plus_examples/handlers.py [hunk 1]
```diff
@@ -771,4 +772,45 @@ def handle_ncreate(event, storage_dir, db_path, cli_config, logger):
...
+            ae_title=globalsubscriber,
+            max_pdu=16382,
+        )
+        
+        if assoc.is_established:
+            try:
+                logger.info(f"Send UPS State Report: {ds.SOPInstanceUID}, {ds.ProcedureStepState}")
+                event_info = Dataset()
+                event_info.ProcedureStepState = ds.ProcedureStepState
+                event_info.InputReadinessState = ds.InputReadinessState
```
Comment (sjswerdloff) [inline tdwii_plus_examples/handlers.py:797]: InputReadinessState is type 1 for SCU and SCP per Table CC.2.5-3. UPS SOP Class N-CREATE/N-SET/N-GET/C-FIND Attributes.

File: tdwii_plus_examples/handlers.py [hunk 2]
```diff
@@ -771,4 +772,45 @@ def handle_ncreate(event, storage_dir, db_path, cli_config, logger):
...
+            contexts=UnifiedProcedurePresentationContexts,
+            ae_title=globalsubscriber,
+            max_pdu=16382,
+        )
+        
+        if assoc.is_established:
+            try:
+                logger.info(f"Send UPS State Report: {ds.SOPInstanceUID}, {ds.ProcedureStepState}")
+                event_info = Dataset()
+                event_info.ProcedureStepState = ds.ProcedureStepState
```
Comment (sjswerdloff) [inline tdwii_plus_examples/handlers.py:796]: CC.2.5.2 Service Class User Behavior:
The SCU shall specify a value of "SCHEDULED" for the Attribute Procedure Step State (0074,1000) in the N-CREATE request
primitive.
so... type 1, even though it isn't listed in Table CC.2.5-3

File: tdwii_plus_examples/TDWII_PPVS_subscriber/ppvs_subscriber_widget.py [hunk 3]
```diff
@@ -62,7 +62,7 @@ def _restart_scp(self):
         self.ppvs_scp.run()
         self.watch_scu = WatchSCU(self.ui.ppvs_ae_line_edit.text())
         ip_addr = "127.0.0.1"
-        port = 11113
+        port = 11114
```
Comment (sjswerdloff) [inline tdwii_plus_examples/TDWII_PPVS_subscriber/ppvs_subscriber_widget.py:65]: need configuration here as well? (for IP address as well)?

**Review by sjswerdloff — APPROVED** [review]
This moves us a good step forward.
Need to add issues for future work as identified in some of the code comments and review comments.

File: tdwii_plus_examples/handlers.py [hunk 4]
```diff
@@ -771,4 +772,45 @@ def handle_ncreate(event, storage_dir, db_path, cli_config, logger):
...
+            11112,
+            contexts=UnifiedProcedurePresentationContexts,
+            ae_title=globalsubscriber,
+            max_pdu=16382,
+        )
+        
+        if assoc.is_established:
+            try:
+                logger.info(f"Send UPS State Report: {ds.SOPInstanceUID}, {ds.ProcedureStepState}")
+                event_info = Dataset()
```
[the remark this answers is not in the document]
Comment (dwikler) [reply tdwii_plus_examples/handlers.py:795]: solved in #32

File: tdwii_plus_examples/handlers.py (hunk 1 above)
Comment (dwikler) [reply tdwii_plus_examples/handlers.py:797]: solved in #32

File: tdwii_plus_examples/handlers.py (hunk 2 above)
Comment (dwikler) [reply tdwii_plus_examples/handlers.py:796]: solved in #32

**Pull request merged**